assets/apps/customizer-controls/src/customizer-search/SearchComponent.tsx

```diff
@@ -0,0 +1,327 @@
+/* global  _wpCustomizeSettings wp */
+
+import React from 'react';
+import { __ } from '@wordpress/i18n';
+import { Button } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Customizer Control Type
+ */
+type Control = {
+	section: string;
+	panelName: string | null;
+	sectionName: string;
+	panel: string;
+	label: string;
+};
+
+/**
+ * Customizer Section Type
+ */
+type Section = {
+	id: string;
+	panel: string;
+	title: string;
+};
+
+/**
+ * Customizer Panel Type
+ */
+type Panel = {
+	id: string;
+	title: string;
+};
+
+/**
+ * Global for TypeScript
+ */
+declare global {
+	const _wpCustomizeSettings: {
+		controls: Control[];
+		sections: Section[];
+		panels: Panel[];
+	};
+
+	const wp: {
+		customize: {
+			section: (sectionName: string) => { expand: () => void };
+		};
+	};
+}
+
+/**
+ * Search Component
+ * Handles search input, filtering and display of results.
+ *
+ * @class
+ */
+const SearchComponent: React.FC = () => {
+	const [search, setSearch] = useState('');
+	const [resultsHTML, setResultsHTML] = useState('');
+	const [isSearchOpen, setIsSearchOpen] = useState(false);
+	const [matchResults, setMatchResults] = useState([] as Control[]);
+
+	const customizerPanels = document.getElementById(
+		'customize-theme-controls'
+	);
+	const searchResults = document.getElementById(
+		'neve-customize-search-results'
+	);
+
+	const controls = Object.values(_wpCustomizeSettings.controls).map(
+		(control) => {
+			Object.values(_wpCustomizeSettings.sections).forEach((section) => {
+				if (control.section === section.id) {
+					Object.values(_wpCustomizeSettings.panels).forEach(
+						(panel) => {
+							if ('' === section.panel) {
+								control.panelName = section.title;
+							}
+
+							if (section.panel === panel.id) {
+								control.sectionName = section.title;
+								control.panel = section.panel;
+								control.panelName = panel.title;
+							}
+						}
+					);
+				}
+			});
+
+			return control;
+		}
+	) as unknown as Control[];
```

Comment:
No need to assert unkonwn here. Maybe a better way to infer the type is to assert it in the map result, so `controls` is inferred as `Control[]`